cards/card_component.js

```diff
@@ -80,6 +113,73 @@ BaseCard["{{componentName}}"] = class extends ANSWERS.Component {
     }
   }
 
+  /**
+   * Triggers the event passed in dispatched from the given selector
+   * @param {string} selector selector to dispatch event from
+   * @param {string} event event to fire
+   * @param {Object} settings additional settings
+   */
+   _triggerCustomEvent(selector, event, settings) {
+    let e = this._customEvent(event, settings);
+    this._container.querySelector(selector).dispatchEvent(e);
+  }
+
+  /**
+   * _customEvent is an event constructor polyfill
+   * @param {string} event event to fire
+   * @param {Object} settings additional settings
+   */
+  _customEvent(event, settings) {
+    const _settings = {
+      bubbles: true,
+      cancelable: true,
+      detail: null,
+      ...settings
+    };
+    const evt = document.createEvent('CustomEvent');
+    evt.initCustomEvent(event, _settings.bubbles, _settings.cancelable, _settings.detail);
+    return evt;
+  }
+
+  /**
+   * reportQuality will send the quality feedback to analytics
+   * @param {boolean} isGood true if the answer is what you were looking for
+   */
+  reportQuality(isGood) {
+    /**
+     * EventTypes are explicit strings defined
+     * for what the server expects for analytics.
+     *
+     * @enum
+     */
+    const EventTypes = {
+      THUMBS_UP: 'THUMBS_UP',
+      THUMBS_DOWN: 'THUMBS_DOWN'
+    };
+    const eventType = isGood === true ? EventTypes.THUMBS_UP : EventTypes.THUMBS_DOWN;
+    const event = new ANSWERS.AnalyticsEvent(eventType)
+      .addOptions({
+        directAnswer: false,
+        experienceKey: this.experienceKey,
+        experienceVersion: this.experienceVersion,
+        queryId: this.queryId,
```

Comment:
I may be mistaken, but I thought that the queryId was automatically sent when analytics are reported? Is that not the behavior you are seeing?

Reply:
Just checked, and that is the case, so I'll remove it from here